### PR TITLE
Add exprfilterprocessor skeleton

### PR DESCRIPTION
**Pull request by pmcollins**
**Description:** Adds an exprfilterprocessor skeleton.

**Testing:** Unit tests added.

### CODE REVIEW

File: processor/labelfilterprocessor/README.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+                label_keys:
+                  - "my-label-key1":
+                      match_type: strict
+                      label_vals:
+                        - "my-label-value1":
+                            match_type: strict
+                        - "my-label-value2":
+                            match_type: strict
+                  - "my-.*-key2":
+                      match_type: regex
```
Comment (keitwb) [inline processor/labelfilterprocessor/README.md:57]: Honestly this is pretty tough on the eyes.

Comment (keitwb) [reply processor/labelfilterprocessor/README.md:57]: What Jay proposed in #1081 is nice.

Also why not treat labels and metric name as siblings instead?  What advantage does it being a hierarchical tree have?

Comment (pmcollins) [reply processor/labelfilterprocessor/README.md:57]: Agree about tough on the eyes. Looking into using expr instead as recommended by Jay. PR is here #1876.

**Comment by pmcollins** [comment]
contribtest failure appears unrelated to this change

**Comment by pmcollins** [comment]
Any thoughts on a name for this package and processor @keitwb and @jrcamp? typeStr `filter` and package `filterprocessor` are taken. Do we want to use `labelfilter`/`labelfilterprocessor` or use something else like `expfilter`/`expfilterprocessor`?

**Comment by keitwb** [comment]
It should also be able to filter on metric name so I would avoid calling it _label_ anything. Also this is basically a datapoint filter in the end right or would it also be expanded to work with logs/traces?

**Comment by pmcollins** [comment]
> It should also be able to filter on metric name so I would avoid calling it _label_ anything. Also this is basically a datapoint filter in the end right or would it also be expanded to work with logs/traces?

Yeah I have the same reservations about `label`. It's just a datapoint filter and I'm not aware of plans to expand to logs or traces, or whether it would useful to have a multi-purpose filter. So... `datapointfilter`? :-) cc: @bogdandrutu @tigrannajaryan

**Comment by pmcollins** [comment]
Renamed to `exprfilter`.

**Comment by rmfitzpatrick** [comment]
Would you be able to provide a readme with what this intends to accomplish/provide?  Is this an implementation of https://github.com/open-telemetry/opentelemetry-collector/issues/560?

**Comment by tigrannajaryan** [comment]
@pmcollins yes, please provide the context of the change, what's the goal. Also, if this is a POC I suggest to move this to contrib repo.

**Comment by pmcollins** [comment]
> Would you be able to provide a readme with what this intends to accomplish/provide? Is this an implementation of #560?
> 
> edit: see it was removed w/ [4ecdce3](https://github.com/open-telemetry/opentelemetry-collector/commit/4ecdce35c103ac2ce5431a3554f0c185589b51b8). I think providing some context in this diff would be helpful though.

Yeah, we're going in a different direction than the proposal in the first readme so I deleted it. The design has been somewhat in flux but I'll capture what we have in a new readme.

**Comment by pmcollins** [comment]
> @pmcollins yes, please provide the context of the change, what's the goal. Also, if this is a POC I suggest to move this to contrib repo.

Have been working with Ben and Jay who know the context, but will add a readme and move to contrib.

**Pull request closed without merging**